centreon/www/front_src/src/Resources/Listing/Listing.cypress.spec.tsx

```diff
@@ -0,0 +1,465 @@
+      where({ information: includes('\n') }),
+      retrievedListing.result
+    );
+    cy.waitFiltersAndListingRequests();
+
+    resourcesWithMultipleLines.forEach(({ information }) =>
+      cy
+        .contains(pipe(split('\n'), head)(information as string) as string)
+        .should('exist')
+    );
+    resourcesWithSingleLines.forEach(({ information }) => {
+      cy.contains(information as string).should('exist');
+    });
+
+    cy.matchImageSnapshot();
+  });
+});
+
+describe('column sorting', () => {
+  const columnToSort = columns
+    .filter(({ sortable }) => sortable !== false)
+    .filter(({ id }) => includes(id, defaultSelectedColumnIds));
+
+  beforeEach(() => {
+    cy.interceptAPIRequest({
+      alias: 'filterRequest',
+      method: Method.GET,
+      path: '**/events-view*',
+      response: fakeData
+    });
+
+    cy.mount({
+      Component: (
+        <Router>
+          <div style={{ backgroundColor: '#fff' }}>
+            <ListingTestWithJotai />
+          </div>
+        </Router>
+      )
+    });
+
+    cy.viewport(1200, 1000);
+
+    columnToSort.forEach(({ id, label, sortField }) => {
+      const sortBy = (sortField || id) as string;
+      const secondSortCriteria =
+        not(equals(sortField, 'last_status_change')) &&
+        defaultSecondSortCriteria;
+
+      const requestUrlDesc = getListingEndpoint({
+        sort: {
+          [sortBy]: 'desc',
+          ...secondSortCriteria
+        }
+      });
+
+      cy.interceptAPIRequest({
+        alias: `dataToListingTableDesc${label}`,
+        method: Method.GET,
+        path: replace('./api/latest/monitoring', '**', requestUrlDesc),
+        response: retrievedListing
+      });
+
+      const requestUrlAsc = getListingEndpoint({
+        sort: {
+          [sortBy]: 'asc',
+          ...secondSortCriteria
+        }
+      });
+      cy.interceptAPIRequest({
+        alias: `dataToListingTableAsc${label}`,
+        method: Method.GET,
+        path: replace('./api/latest/monitoring', '**', requestUrlAsc),
+        response: retrievedListing
+      });
+    });
+  });
+
+  columnToSort.forEach(({ label }) => {
+    it(`executes a listing request with sort_by param and stores the order parameter in the URL when ${label} column is clicked`, () => {
+      cy.waitForRequest('@filterRequest');
+
+      cy.findByLabelText(`Column ${label}`).click();
+
+      cy.waitForRequest(`@dataToListingTableDesc${label}`);
+
+      cy.findByLabelText(`Column ${label}`).click();
+
+      cy.waitForRequest(`@dataToListingTableAsc${label}`);
+
+      cy.matchImageSnapshot();
+    });
+  });
+});
+
+describe('Listing request', () => {
+  beforeEach(() => {
+    interceptRequestsAndMountBeforeEach();
+  });
+
+  it('executes a listing request with an updated page param when a change page action is clicked', () => {
+    cy.waitFiltersAndListingRequests();
+
+    cy.findByLabelText(`Next page`)
+      .should(($label) => {
```

Comment:
```suggestion
      .should((label) => {
```
Avoid "$" on variable name as it is not conventional in JavaScript